src/solid_poro_3D_ele/4C_solid_poro_3D_ele_calc_pressure_velocity_based.cpp

```diff
@@ -330,42 +369,178 @@ void Discret::Elements::SolidPoroPressureVelocityBasedEleCalc<
 
           update_geometric_stiffness_matrix<celltype>(sfac, jacobian_mapping.N_XYZ_, *stiff);
 
-          if (react.has_value())
+          if (react_matrix.has_value())
           {
             /* additional "reactive darcy-term"
             detJ * w(gp) * ( J * reacoeff * phi^2  ) * D(v_s)
   */
-            react->update(1.0, erea_v, 1.0);
+            react_matrix->update(1.0, erea_v, 1.0);
           }
         }
       });
 }
 
-template <Core::FE::CellType celltype>
-void Discret::Elements::SolidPoroPressureVelocityBasedEleCalc<
-    celltype>::evaluate_nonlinear_force_stiffness_od(const Core::Elements::Element& ele,
+template <Core::FE::CellType celltype, Discret::Elements::PorosityFormulation porosity_formulation>
+void Discret::Elements::SolidPoroPressureVelocityBasedEleCalc<celltype, porosity_formulation>::
+    evaluate_nonlinear_force_stiffness_porosity_od(const Core::Elements::Element& ele,
+        Mat::StructPoro& porostructmat, Mat::FluidPoro& porofluidmat,
+        AnisotropyProperties anisotropy_properties, const Inpar::Solid::KinemType& kinematictype,
+        const Core::FE::Discretization& discretization,
+        const SolidPoroPrimaryVariables<porosity_formulation>& primary_variables,
+        const std::vector<double>& initial_porosity, Teuchos::ParameterList& params,
+        Core::LinAlg::SerialDenseVector* porosity_force_vector_inp,
+        Core::LinAlg::SerialDenseMatrix* K_pp_inp, Core::LinAlg::SerialDenseMatrix* K_dp_inp,
+        Core::LinAlg::SerialDenseMatrix* K_pd_inp)
+{
+  constexpr int num_poro_dof = Core::FE::num_nodes<celltype>;
+  constexpr int num_disp_dof = Core::FE::num_nodes<celltype> * Core::FE::dim<celltype>;
+
+  auto K_pp = make_optional_matrix_view<num_poro_dof, num_poro_dof>(K_pp_inp);
```

Comment:
```suggestion
  auto K_pp = make_optional_matrix_view<num_porosity_dof, num_porosity_dof>(K_pp_inp);
```
For me, `poro` was not super clear at the first sight.